src/ThreeEditor/js/object/Object.Grid.ts

```diff
@@ -0,0 +1,115 @@
+//GRID DENSITY
+//WIDTH DENSITY
+//HEIGHT DENSITY
+//DEPTH DENSITY
+
+import { DetectGeometry } from '../../util/Detect/DetectGeometry';
+import { createRowParamNumber, hideUIElement, showUIElement } from '../../util/UiUtils';
+import { SetDetectGeometryCommand } from '../commands/Commands';
+import { Editor } from '../Editor';
+import { UINumber, UIRow } from '../libs/ui';
+import { ObjectAbstract } from './Object.Abstract';
+
+//RADIAL DENSITY
+export class ObjectGrid extends ObjectAbstract {
+	object?: DetectGeometry;
+
+	widthRow: UIRow;
```

Comment:
I'd suggest to completely get rid of width, height and depth naming and replace it with X,Y,Z-related names which are much less confusing

Reply:
I've replaced some of them in the new files, but the rest should be done separately so that it will be easier to debug potential problems. #306